### PR TITLE
Assets xcm

### PR DESCRIPTION
**Pull request by girazoki**
Adds pallet-assets, foreign-asset-creator and asset-rate as necessary pallets to mint derivative tokens coming from other chains, with these respective goals:

- pallet-assets: handle representation and assets of derivative tokens
- foreign-asset-creator: maintain a mapping between MultiLocation and assetId, which should make it easier to handle future ML versions.
- asset-rate: maintains the rate at which we can charge weight execution fees for certain tokens

TODO:
- update benchmakrs

For future PRs:

- Incorporate these into templates

### CODE REVIEW

**Review by Agusrodri — APPROVED** [review]
I didn't see anything that could block the PR. Although I think we should keep an eye on the runtime code that could be moved to polkadot-sdk, like the one related to `NativeAssetReserve` for instance. Also a follow up task would be to add some TS tests (the rust ones look great). Other than those comments LGTM!

**Comment by girazoki** [comment]
thank Agustin! Yeah I will add a few more tests in TS for sure before I merge this PR

File: test/suites/dev-tanssi/xcm-token-reception/test_dmp_token_reception _appendix.ts
Comment (tmpolaczyk) [inline test/suites/dev-tanssi/xcm-token-reception/test_dmp_token_reception _appendix.ts:1]: Some of these files have a space in its filename, can you rename them?

Comment (girazoki) [reply test/suites/dev-tanssi/xcm-token-reception/test_dmp_token_reception _appendix.ts:1]: yes!

**Review by fgamundi — APPROVED** [review]

**Comment by girazoki** [comment]
We need to update the benchmark for buy_execution but it is dependent on `https://github.com/paritytech/polkadot-sdk/pull/2962`. So for now I think we can postpone this for a future PR

**Pull request merged**
